pkg/virt-controller/watch/network_test.go

```diff
@@ -167,36 +169,89 @@ var _ = Describe("Network interface hot{un}plug", func() {
+			vmi.Spec.Domain.Devices.Interfaces = vmiIfaces
+			vm := VirtualMachineFromVMI("test", vmi, true)
+			vm.Spec.Template.Spec.Domain.Devices.Interfaces = vmIfaces
 			vm.Status.InterfaceRequests = ifaceRequests
 
-			trimDoneInterfaceRequests(vm)
+			trimDoneInterfaceRequests(vm, vmi)
 
 			Expect(vm.Status.InterfaceRequests).To(Equal(expectedIfaceRequests))
 		},
 		Entry("have request removed on successful hotplug",
+			[]v1.Interface{{Name: "blue"}},
 			[]v1.Interface{{Name: "blue"}},
 			[]v1.VirtualMachineInterfaceRequest{{AddInterfaceOptions: &v1.AddInterfaceOptions{Name: "blue"}}},
 			[]v1.VirtualMachineInterfaceRequest{},
 		),
 		Entry("keep interface request for pending hotplug",
+			[]v1.Interface{},
+			[]v1.Interface{},
+			[]v1.VirtualMachineInterfaceRequest{{AddInterfaceOptions: &v1.AddInterfaceOptions{Name: "blue"}}},
+			[]v1.VirtualMachineInterfaceRequest{{AddInterfaceOptions: &v1.AddInterfaceOptions{Name: "blue"}}},
+		),
+		Entry("keep interface request for pending hotplug, when VMI spec has not been updated yet",
+			[]v1.Interface{{Name: "blue"}},
 			[]v1.Interface{},
 			[]v1.VirtualMachineInterfaceRequest{{AddInterfaceOptions: &v1.AddInterfaceOptions{Name: "blue"}}},
 			[]v1.VirtualMachineInterfaceRequest{{AddInterfaceOptions: &v1.AddInterfaceOptions{Name: "blue"}}},
 		),
+		Entry("keep interface request for pending hotplug, when VM template spec has not been updated yet",
+			[]v1.Interface{},
+			[]v1.Interface{{Name: "blue"}},
+			[]v1.VirtualMachineInterfaceRequest{{AddInterfaceOptions: &v1.AddInterfaceOptions{Name: "blue"}}},
+			[]v1.VirtualMachineInterfaceRequest{{AddInterfaceOptions: &v1.AddInterfaceOptions{Name: "blue"}}},
+		),
 		Entry("have request removed on successful unplug",
+			[]v1.Interface{{Name: "blue", State: v1.InterfaceStateAbsent}},
 			[]v1.Interface{{Name: "blue", State: v1.InterfaceStateAbsent}},
 			[]v1.VirtualMachineInterfaceRequest{{RemoveInterfaceOptions: &v1.RemoveInterfaceOptions{Name: "blue"}}},
 			[]v1.VirtualMachineInterfaceRequest{},
 		),
 		Entry("keep interface request for pending unplug",
 			[]v1.Interface{{Name: "blue"}},
+			[]v1.Interface{{Name: "blue"}},
+			[]v1.VirtualMachineInterfaceRequest{{RemoveInterfaceOptions: &v1.RemoveInterfaceOptions{Name: "blue"}}},
+			[]v1.VirtualMachineInterfaceRequest{{RemoveInterfaceOptions: &v1.RemoveInterfaceOptions{Name: "blue"}}},
+		),
+		Entry("keep interface request for pending unplug, when VMI spec has not been updated yet",
+			[]v1.Interface{{Name: "blue"}},
+			[]v1.Interface{{Name: "blue", State: v1.InterfaceStateAbsent}},
 			[]v1.VirtualMachineInterfaceRequest{{RemoveInterfaceOptions: &v1.RemoveInterfaceOptions{Name: "blue"}}},
 			[]v1.VirtualMachineInterfaceRequest{{RemoveInterfaceOptions: &v1.RemoveInterfaceOptions{Name: "blue"}}},
 		),
+		Entry("keep interface request for pending unplug, when VM template spec has not been updated yet",
+			[]v1.Interface{{Name: "blue", State: v1.InterfaceStateAbsent}},
```

Comment:
Same, seem opposite to the test text.

And this state is also in the gray area, I am not sure it should keep the command.
In the end, it is an invalid or undefined state.

Reply:
Same